tests/e2e/local/common.sh

```diff
@@ -0,0 +1,144 @@
+function check_apt_get() {
+  if ! apt-get --help > /dev/null; then
+    echo "apt-get not installed. Please install it and run this script again."
+    exit 1
+fi
+
+# Install Docker
+function install_docker_debian() {
+  echo "Checking and Installing Docker as required"
+  if ! docker --help > /dev/null; then
+    # docker-ce depends on libltdl7
+    apt-get install -y libltdl7
+    curl -L https://download.docker.com/linux/debian/dists/stretch/pool/stable/amd64/docker-ce_18.03.0~ce-0~debian_amd64.deb -o docker-ce.deb
+    if ! sudo dpkg -i docker-ce.deb; then
+      echo "Looks like docker installation failed."
+      echo "Please install it manually and then run this script again."
+      exit 1
+    fi
+  fi
+}
+
+function install_docker_macos() {
+  echo "Checking docker..."
+  if ! docker --help > /dev/null; then
+    echo "docker is not installed. Install it from homebrew cask."
+    if ! brew cask install docker; then
+      echo "Installation of docker from brew fails. Please install it manually."
+      exit 1
+    else
+      echo "Done."
+    fi
+  else
+    echo "docker exists. Please make sure to update it to latest version."
+  fi
+}
+
+function install_docker_machine() {
+  echo "Checking docker-machine..."
+  if ! docker-machine --help > /dev/null; then
+    echo "docker-machine is not installed. Downloading and Installing it using curl."
+    base=https://github.com/docker/machine/releases/download/v0.14.0 &&
+    if ! curl -L "$base/docker-machine-$(uname -s)-$(uname -m)" >/usr/local/bin/docker-machine && chmod +x /usr/local/bin/docker-machine; then
+        echo "Installation of docker-machine failed. Please install it manually."
+        exit 1
+    else
+        echo "Done."
+    fi
+  else
+    echo "docker-machine exists. Please make sure to update it to latest version."
+    docker-machine version
+  fi
+}
+
+# Install Kubectl on Linux platform
+function install_kubectl_linux() {
+  echo "Checking and Installing Kubectl as required"
+  if ! kubectl --help > /dev/null; then
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/"$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)"/bin/linux/amd64/kubectl
+    chmod +x ./kubectl
+    sudo mv ./kubectl /usr/local/bin/kubectl
+  fi
+}
+
+function install_kubectl_macos() {
+  echo "Checking kubectl..."
+  if ! kubectl --help > /dev/null; then
+    echo "kubectl is not installed. Installing the lastest stable release..."
+    if ! brew install kubectl; then
+    	echo "Installation of kubectl from brew fails. Please install it manually."
+        exit 1
+    else
+    	echo "Done."
+    fi
+  else
+    echo "kubectl exists. Please make sure to update it to latest version."
+  fi
+}
+
+# Check if homebrew is installed
+function check_homebrew() {
+  if ! brew --help > /dev/null; then
+    echo "Homebrew is not installed. Please go to https://docs.brew.sh/Installation to install Homebrew."
+    exit 1
+  fi
+}
```

Comment:
A single function like check_install_cmd() with two arguments 'cmd' and 'install_url", should cover brew, apt-get, dpkg.